core/dbt/events/functions.py

```diff
@@ -127,9 +128,32 @@ def create_text_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> str:
 # translates an Event to a completely formatted json log line
 # you have to specify which message you want. (i.e. - e.message, e.cli_msg(), e.file_msg())
 def create_json_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> str:
-    values = e.to_dict(scrub_secrets(msg_fn(e), env_secrets()))
+    values = e.to_dict(msg_fn(e))
     values['ts'] = e.ts.isoformat()
-    log_line = json.dumps(values, sort_keys=True)
+    if hasattr(e, '__dataclass_fields__'):
+        values['data'] = {
+            x: getattr(e, x) for x, y
+            in e.__dataclass_fields__.items()  # type: ignore[attr-defined]
+            if type(y._field_type) == _FIELD_BASE
+        }
+    else:
+        values['data'] = None
+
+    # need to catch if any data is not serializable but still make sure as much of
+    # the logs go out as possible
+    try:
+        log_line = json.dumps(
+            {k: scrub_secrets(v, env_secrets()) for (k, v) in values.items()},
+            sort_keys=True
+        )
+    except TypeError:
+        # the only key currently throwing errors is 'data'.  Expand this list
+        # as needed if new issues pop up
+        safe_values = {k: v for (k, v) in values.items() if k not in ('data')}
```

Comment:
Since you're not using `values` after this point you can just pop the `data` key.  
```
except...
    values.pop("data")
    log_line = blahblah(in values.items())
 ```